
Imported Dependencies for Analysis


In [95]:
#imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt 
from sklearn.cluster import KMeans 
import plotly.express as px
from sklearn.decomposition import PCA
import numpy as np

from sklearn.cluster import DBSCAN
from sklearn import metrics
from sklearn.datasets import make_blobs
from sklearn.preprocessing import StandardScaler

Load the csv file and show top 5 records from it

In [96]:
df = pd.read_csv('/content/Video_Games_Sales_as_at_22_Dec_2016.csv')
df.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count  User_Score  User_Count Developer Rating  
0          51.0         8.0       322.0  Nintendo      E  
1           NaN         NaN         NaN       NaN    NaN  
2          73.0         8.3       709.0  Nintendo      E  
3          73.0         8.0       192.0  Nintendo      E  
4           NaN         NaN         NaN       NaN    NaN

In [97]:
#returns number of duplicate values in dataframe
print("Total amount of duplicated records in dataset: ", df.duplicated().sum())

Total amount of duplicated records in dataset:  0


There is no duplicated data at all

In [98]:
print("Total Number of Rows in Dataset: ", len(df))

Total Number of Rows in Dataset:  16719


In [99]:
print("Total Number of Columns in Dataset: ", len(df.columns))

Total Number of Columns in Dataset:  16


In [100]:
#returns number of duplicate values in dataframe
df.duplicated().sum()

0

There are no duplicate values.

Checking for null values in dataset

In [101]:
df.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         9129
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

Alot of missing data here.

In [102]:
for (columnName, columnData) in df.iteritems():
    print('Does Column have Null, N/A, NAN values', columnName, "True or False?: ",df[columnName].isnull().values.any())

Does Column have Null, N/A, NAN values Name True or False?:  True
Does Column have Null, N/A, NAN values Platform True or False?:  False
Does Column have Null, N/A, NAN values Year_of_Release True or False?:  True
Does Column have Null, N/A, NAN values Genre True or False?:  True
Does Column have Null, N/A, NAN values Publisher True or False?:  True
Does Column have Null, N/A, NAN values NA_Sales True or False?:  False
Does Column have Null, N/A, NAN values EU_Sales True or False?:  False
Does Column have Null, N/A, NAN values JP_Sales True or False?:  False
Does Column have Null, N/A, NAN values Other_Sales True or False?:  False
Does Column have Null, N/A, NAN values Global_Sales True or False?:  False
Does Column have Null, N/A, NAN values Critic_Score True or False?:  True
Does Column have Null, N/A, NAN values Critic_Count True or False?:  True
Does Column have Null, N/A, NAN values User_Score True or False?:  True
Does Column have Null, N/A, NAN values User_Count True or False?: 

Please Refer to heatmap located on notebook: data_project_4_pandas_profile_phase1_part2.ipynb

Pandas Profile generated a heatmap of the missing data. It should be located in submitted zipped folder.

Drop Useless Columns from Dataset

In [103]:
df.drop(['Critic_Count', 'User_Count'], axis = 1, inplace = True) 

In [104]:
print("Total Number of Columns in Dataset: ", len(df.columns))

Total Number of Columns in Dataset:  14


I dropped user and critic count becuase it gives no useful purpose for analysis and meaning of that column is unclear.

Dropping Null Values 

In [105]:
print("Percentage of Missing data within dataset: ", round(df.isnull().sum() * 100 / len(df), 2))

Percentage of Missing data within dataset:  Name                0.01
Platform            0.00
Year_of_Release     1.61
Genre               0.01
Publisher           0.32
NA_Sales            0.00
EU_Sales            0.00
JP_Sales            0.00
Other_Sales         0.00
Global_Sales        0.00
Critic_Score       51.33
User_Score         54.60
Developer          39.61
Rating             40.49
dtype: float64


In [106]:
# replace rows with null NAN N/A values
df["Name"].fillna("No Name",inplace=True)
df["Year_of_Release"].fillna(0000,inplace=True)
df["Genre"].fillna("No Genre",inplace=True)
df["Publisher"].fillna("No Publisher",inplace=True)
df["Critic_Score"].fillna(0.0,inplace=True)
df["User_Score"].fillna(0.0,inplace=True)
df["Developer"].fillna("No Developer",inplace=True)
df["Rating"].fillna("No Rating",inplace=True)

In [107]:
# Convert from float to year
df['Year_of_Release'] = df['Year_of_Release'].astype(int)

In [108]:
df.head(2)

Name Platform  Year_of_Release     Genre Publisher  NA_Sales  \
0         Wii Sports      Wii             2006    Sports  Nintendo     41.36   
1  Super Mario Bros.      NES             1985  Platform  Nintendo     29.08   

   EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  User_Score  \
0     28.96      3.77         8.45         82.53          76.0         8.0   
1      3.58      6.81         0.77         40.24           0.0         0.0   

      Developer     Rating  
0      Nintendo          E  
1  No Developer  No Rating

Changed column: 'Year_of_Release' from float to integer.

In [109]:
print("Total Number of Rows in Dataset: ", len(df))

Total Number of Rows in Dataset:  16719


Total Rows: 16719 - Total Rows left after nulls dropped: 6825 = Total Rows removed with null values: 9894 

In [110]:
df.isnull().sum()

Name               0
Platform           0
Year_of_Release    0
Genre              0
Publisher          0
NA_Sales           0
EU_Sales           0
JP_Sales           0
Other_Sales        0
Global_Sales       0
Critic_Score       0
User_Score         0
Developer          0
Rating             0
dtype: int64

In [111]:
#returns number of duplicate values in dataframe
print("Total amount of duplicated records in dataset: ", df.duplicated().sum())

Total amount of duplicated records in dataset:  0


All of the dataset is cleaned with no duplicated or null values.

Originally, if I had removed all rows with null values over half of the dataset would be dropped as a result. Percentage of null/NaN rows removed was 59.18 %. So instead of dropping them, I just replaced them with String or numeric data "No Data" or 0.0 or 0000 for the release year. Sales columns had no Null values.

Export Cleaned to data to csv format

In [112]:
df.to_csv('Video_Games_Sales_as_at_22_Dec_2016_cleaned.csv')